public/app/features/alerting/unified/hooks/useExternalAmSelector.ts

```diff
@@ -1,76 +1,119 @@
-import { countBy, keyBy } from 'lodash';
-import { createSelector } from 'reselect';
+import { useEffect } from 'react';
 
-import { DataSourceInstanceSettings, DataSourceJsonData, DataSourceSettings } from '@grafana/data';
-import { AlertManagerDataSourceJsonData } from 'app/plugins/datasource/alertmanager/types';
-import { StoreState, useSelector } from 'app/types';
+import { DataSourceSettings } from '@grafana/data';
+import { AlertManagerDataSourceJsonData, ExternalAlertmanagers } from 'app/plugins/datasource/alertmanager/types';
 
 import { alertmanagerApi } from '../api/alertmanagerApi';
-import { getAlertManagerDataSources } from '../utils/datasource';
+import { dataSourcesApi } from '../api/dataSourcesApi';
+import { isAlertmanagerDataSource } from '../utils/datasource';
 
-export interface ExternalDataSourceAM {
-  dataSource: DataSourceInstanceSettings<AlertManagerDataSourceJsonData>;
-  url?: string;
-  status: 'active' | 'pending' | 'dropped';
-  statusInconclusive?: boolean;
-}
-
-export function useExternalDataSourceAlertmanagers(): ExternalDataSourceAM[] {
-  const { useGetExternalAlertmanagersQuery } = alertmanagerApi;
-  const { currentData: discoveredAlertmanagers } = useGetExternalAlertmanagersQuery();
+type ConnectionStatus = 'active' | 'pending' | 'dropped' | 'inconclusive' | 'uninterested' | 'unknown';
 
-  const externalDsAlertManagers = getAlertManagerDataSources().filter((ds) => ds.jsonData.handleGrafanaManagedAlerts);
+export interface ExternalAlertmanagerDataSourceWithStatus {
+  dataSourceSettings: DataSourceSettings<AlertManagerDataSourceJsonData>;
+  status: ConnectionStatus;
+}
 
-  const alertmanagerDatasources = useSelector(
-    createSelector(
-      (state: StoreState) => state.dataSources.dataSources.filter((ds) => ds.type === 'alertmanager'),
-      (datasources) => keyBy(datasources, (ds) => ds.uid)
-    )
-  );
+/**
+ * Returns all configured Alertmanager data sources and their connection status with the internal ruler
+ */
+export function useExternalDataSourceAlertmanagers(): ExternalAlertmanagerDataSourceWithStatus[] {
+  // firstly we'll fetch the settings for all datasources and filter for "alertmanager" type
+  const [fetchDataSourceSettings, { alertmanagerDataSources }] =
+    dataSourcesApi.endpoints.getDataSourceSettings.useLazyQuery({
+      refetchOnReconnect: true,
+      refetchOnFocus: true,
+      selectFromResult: (result) => {
+        const alertmanagerDataSources = result.currentData?.filter(isAlertmanagerDataSource) ?? [];
+        return { ...result, alertmanagerDataSources };
+      },
+    });
+
+  // we'll also fetch the configuration for which Alertmanagers we are forwarding Grafana-managed alerts too
+  // @TODO use polling when we have one or more alertmanagers in pending state
```

Comment:
I don't have any good ideas here, I tried a few things with `setState` and tracking pending items but got stuck in infinite render loops 😅  ideas?

Reply:
why do we need this poll?

Reply:
When a user adds an Alertmanager to receive alerts it will be "adopted" by the built-in Alertmanager as a peer. 

This process takes a while, so ideally we would poll the `/api/v1/ngalert/alertmanagers` endpoint until it settles on either `dropped` (when unreachable) or `active` (peer added).

If we don't then we have to put the onus on the user to refresh the admin page periodically to see the latest connection state.